app/goal_routes.py

```diff
@@ -0,0 +1,126 @@
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+from flask import Blueprint, jsonify, make_response, request, abort
+import requests
+
+
+
+goal_bp = Blueprint("goal_bp", __name__, url_prefix='/goals')
+
+@goal_bp.route("", methods=["POST"])
+def create_goal():        
+    request_body = request.get_json()
+    if "title" not in request_body:
+        return make_response(jsonify({"details":"Invalid data"})),400
+    new_goal= Goal(
+        title = request_body["title"]
+    )
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return {"goal": {
+                "id": new_goal.goal_id,
+                "title": new_goal.title
+            }},201
+
+@goal_bp.route("", methods=["GET"])
+def read_all_goals():
+    goals = Goal.query.all()
+    
+    goals_response = []
+    for goal in goals:
+        goals_response.append({
+            "id": goal.goal_id,
+            "title": goal.title
+            })
+    
+    return jsonify(goals_response),200
+
+def validate_goal(goal_id):
+    try:
+        goal_id = int(goal_id)
+    except:
+        abort(make_response({"details": "Invalid data"}, 400))
+
+    goal = Goal.query.get(goal_id)
+
+    if not goal:
+        abort(make_response(jsonify(dict(details=f"There is no existing goal {goal_id}")), 404))
+
+    return goal
```

Comment:
Great choice creating this helper function. Typically helper methods are placed in a separate file like routes_helpers. This just helps keep our model clean and separates our concerns